### PR TITLE
Deprecate process -v/--value in favor of -p/--param

### LINKED ISSUE #8577: `oc new-app -p` is inconsistent with `oc process -v`
`oc new-app` and `oc process` use a different flag name for the same concept of supplying template parameters. Since they both do the same thing with templates, this should be consistent.
##### Version

oc v1.1.6-88-gbaf4504
kubernetes v1.2.0-36-g4a3f9c5
##### Steps To Reproduce
1. `oc new-app -h |& grep pairs`
2. `oc process -h |& grep pairs`
##### Current Result

```
$ oc new-app -h |& grep pairs
  -e, --env=[]: Specify key value pairs of environment variables to set into each container.
  -p, --param=[]: Specify a list of key value pairs (e.g., -p FOO=BAR,BAR=FOO) to set/override parameter values in the template.
$ oc process -h |& grep pairs                                                                                                                                                       
  -v, --value=[]: Specify a list of key-value pairs (eg. -v FOO=BAR,BAR=FOO) to set/override parameter values
```
##### Expected Result

```
$ oc new-app -h |& grep pairs
  -e, --env=[]: Specify key value pairs of environment variables to set into each container.
  -v, --value=[]: Specify a list of key value pairs (e.g., -p FOO=BAR,BAR=FOO) to set/override parameter values in the template.
```

(I suggest deprecating -p and using -v as `oc process` was around first)

### PR DESCRIPTION
**Pull request by jupierce**
Fixes #8577
ptal @bparees @csrwng 
Assuming this is acceptable, I will make a corresponding PR for openshift-docs.

### CODE REVIEW

**Comment by bparees** [comment]
i'm certainly in favor of it.  @fabianofranz any issues w/ this?

**Comment by bparees** [comment]
@openshift/api-review ptal

**Comment by smarterclayton** [comment]
Test case showing deprecated behavior works.

API change approved.

**Comment by jupierce** [comment]
[test]

**Comment by fabianofranz** [comment]
> any issues w/ this?

Works for me.

File: test/cmd/process.sh [hunk 1]
```diff
@@ -43,29 +48,40 @@ os::cmd::expect_failure_and_text "oc process -f '${OS_ROOT}/test/testdata/basic-
 # not providing required parameter should fail
 os::cmd::expect_failure_and_text "oc process -f '${required_params}'" 'parameter required_param is required and must be specified'
 # not providing an optional param is OK
-os::cmd::expect_success "oc process -f '${required_params}' --value=required_param=someval | oc create -f -"
+os::cmd::expect_success "oc process -f '${required_params}' --value=required_param=someval"
+os::cmd::expect_success "oc process -f '${required_params}' -v required_param=someval"
+os::cmd::expect_success "oc process -f '${required_params}' --param=required_param=someval"
+os::cmd::expect_success "oc process -f '${required_params}' -p required_param=someval | oc create -f -"
 # parameters with multiple equal signs are OK
 os::cmd::expect_success "oc process -f '${required_params}' required_param=someval=moreval | oc create -f -"
```
Comment (jupierce) [inline test/cmd/process.sh]: Will do.

Comment (jupierce) [inline test/cmd/process.sh]: @deads2k 27067a9 ptal

Comment (deads2k) [inline test/cmd/process.sh]: > @deads2k 27067a9 ptal

that works, thanks.

**Comment by jupierce** [comment]
flake #11016   https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_conformance/8842/consoleFull

**Comment by bparees** [comment]
[merge]

**Comment by jupierce** [comment]
flake #11016  https://ci.openshift.redhat.com/jenkins/job/test_pr_origin/11665/
[test]

**Comment by bparees** [comment]
[merge]

**Pull request merged**